### PR TITLE
changed: put the network class in the service manager

### PR DESCRIPTION
**Pull request by notspiff**
and expose through the service broker. allows removing some Application.h includes

### CODE REVIEW

**Comment by Rechi** [comment]
https://github.com/xbmc/xbmc/blob/b1cafa83f67b90f0cd4ada2217dad9013665d118/xbmc/test/TestBasicEnvironment.cpp#L64-L65
has to be changed

File: xbmc/ServiceManager.cpp [hunk 1]
```diff
@@ -207,6 +209,7 @@ void CServiceManager::DeinitStageOne()
   m_XBPython.reset();
 #endif
   m_announcementManager.reset();
+  m_network.reset();
```
Comment (garbear) [inline xbmc/ServiceManager.cpp]: Can you place this at the beginning of the function to make sure the destruction order is opposite initialization?

Comment (notspiff) [reply xbmc/ServiceManager.cpp]: while the previous code shows that there is no order issues, it's good to be consistent. changed.

**Comment by notspiff** [comment]
test env adjusted.

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -6762,7 +6762,7 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
     {
       std::string linkStatus = g_localizeStrings.Get(151);
       linkStatus += " ";
-      CNetworkInterface* iface = g_application.getNetwork().GetFirstConnectedInterface();
+      CNetworkInterface* iface = g_application.m_ServiceManager->GetNetwork().GetFirstConnectedInterface();
```
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: this should be `CServiceBroker::GetNetwork()` for all calls in this file.

File: xbmc/test/TestBasicEnvironment.cpp [hunk 3]
```diff
@@ -61,9 +61,6 @@ void TestBasicEnvironment::SetUp()
    */
   g_advancedSettings.Initialize();
 
-  // Need to configure the network as some tests access the network member
-  g_application.SetupNetwork();
```
Comment (xhaggi) [inline xbmc/test/TestBasicEnvironment.cpp]: should this be replaced by `CServiceBroker::GetNetwork().SetupNetwork()`?

File: xbmc/windows/GUIMediaWindow.cpp [hunk 4]
```diff
@@ -1124,7 +1124,7 @@ bool CGUIMediaWindow::HaveDiscOrConnection(const std::string& strPath, int iDriv
   else if (iDriveType==CMediaSource::SOURCE_TYPE_REMOTE)
   {
     //! @todo Handle not connected to a remote share
-    if ( !g_application.getNetwork().IsConnected() )
+    if ( !CServiceBroker::GetNetwork().IsConnected() )
```
Comment (xhaggi) [inline xbmc/windows/GUIMediaWindow.cpp]: please remove spaces if you touch the line

File: xbmc/windows/GUIWindowFileManager.cpp [hunk 5]
```diff
@@ -672,7 +672,7 @@ bool CGUIWindowFileManager::HaveDiscOrConnection( std::string& strPath, int iDri
   else if ( iDriveType == CMediaSource::SOURCE_TYPE_REMOTE )
   {
     //! @todo Handle not connected to a remote share
-    if ( !g_application.getNetwork().IsConnected() )
+    if ( !CServiceBroker::GetNetwork().IsConnected() )
```
Comment (xhaggi) [inline xbmc/windows/GUIWindowFileManager.cpp]: same here please

File: xbmc/windows/GUIWindowLoginScreen.cpp [hunk 6]
```diff
@@ -283,7 +283,7 @@ void CGUIWindowLoginScreen::LoadProfile(unsigned int profile)
 
   if (profile != 0 || !CProfilesManager::GetInstance().IsMasterProfile())
   {
-    g_application.getNetwork().NetworkMessage(CNetwork::SERVICES_DOWN,1);
+    CServiceBroker::GetNetwork().NetworkMessage(CNetwork::SERVICES_DOWN,1);
```
Comment (xhaggi) [inline xbmc/windows/GUIWindowLoginScreen.cpp]: Space between params please

File: xbmc/windows/GUIWindowLoginScreen.cpp [hunk 7]
```diff
@@ -292,7 +292,7 @@ void CGUIWindowLoginScreen::LoadProfile(unsigned int profile)
     if (pWindow)
       pWindow->ResetControlStates();
   }
-  g_application.getNetwork().NetworkMessage(CNetwork::SERVICES_UP,1);
+  CServiceBroker::GetNetwork().NetworkMessage(CNetwork::SERVICES_UP,1);
```
Comment (xhaggi) [inline xbmc/windows/GUIWindowLoginScreen.cpp]: same here please

File: xbmc/GUIInfoManager.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/GUIInfoManager.cpp]: done. there is surrounding code using the form i chose earlier..

File: xbmc/test/TestBasicEnvironment.cpp (hunk 3 above)
Comment (notspiff) [reply xbmc/test/TestBasicEnvironment.cpp]: no.

File: xbmc/windows/GUIMediaWindow.cpp (hunk 4 above)
Comment (notspiff) [reply xbmc/windows/GUIMediaWindow.cpp]: done.

File: xbmc/windows/GUIWindowFileManager.cpp (hunk 5 above)
Comment (notspiff) [reply xbmc/windows/GUIWindowFileManager.cpp]: done.

File: xbmc/windows/GUIWindowLoginScreen.cpp (hunk 6 above)
Comment (notspiff) [reply xbmc/windows/GUIWindowLoginScreen.cpp]: done.

File: xbmc/windows/GUIWindowLoginScreen.cpp (hunk 7 above)
Comment (notspiff) [reply xbmc/windows/GUIWindowLoginScreen.cpp]: done.

File: xbmc/test/TestBasicEnvironment.cpp (hunk 3 above)
Comment (xhaggi) [reply xbmc/test/TestBasicEnvironment.cpp]: there are no network related tests?

**Review by xhaggi — APPROVED** [review]
Looks good .. thx.

File: xbmc/test/TestBasicEnvironment.cpp (hunk 3 above)
Comment (notspiff) [reply xbmc/test/TestBasicEnvironment.cpp]: Network is brought up at service manager init level 1 which is called above.

**Comment by notspiff** [comment]
wayland failure is cryptic to me. is there some newfangled stuff checking for new warnings? did i trigger more 15 new warnings for the wayland build?

**Comment by Rechi** [comment]
Yes there is a warning checker. LINUX-64-Wayland usually produced 14 warnings (which is set as maximum allowed warnings).
With a recent change (not yours) an additional warning was introduced, but Jenkins report was ignored and the PR got merged.

**Comment by notspiff** [comment]
okay, so then i can safely merge this one. thx.

**Pull request merged**